truth_set/python/structure/solution-approach/semantic-duplication/services/userService.py

```diff
@@ -0,0 +1,19 @@
+from datetime import datetime
+from ..utils.dateUtils import format_date
+
+class UserService:
+    def __init__(self):
+        self.users = []
+    
+    def add_user(self, name):
+        new_user = {
+            'id': len(self.users) + 1,
+            'name': name,
+            'registration_date': datetime.now()
+        }
+        self.users.append(new_user)
+        print(f"User Added: {new_user['name']}, Registered On: {format_date(new_user['registration_date'], 'MM/DD/YYYY')}")
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider abstracting common service functionality</b></div>
<div id="fix">

The `add_user` and `add_order` methods appear to have similar functionality for creating and storing records. Consider creating a common base service class with shared functionality. A similar issue was also found in truth_set/python/structure/solution-approach/semantic-duplication/index.py (line 1-17).
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```
 @@ -1,6 +1,24 @@
 +class BaseService:
 +    def __init__(self):
 +        self.records = []
 +
 +    def add_record(self, data, record_type):
 +        new_record = {
 +            'id': len(self.records) + 1,
 +            **data,
 +            'created_at': datetime.now()
 +        }
 +        self.records.append(new_record)
 +        return new_record
 +
 +class UserService(BaseService):
 +    def add_user(self, name):
 +        record = self.add_record({'name': name}, 'user')
 +        print(f"User Added: {record['name']}, Registered On: {format_date(record['created_at'], 'MM/DD/YYYY')}")
 +
 +class OrderService(BaseService):
 +    def add_order(self, product):
 +        record = self.add_record({'product': product}, 'order')
 +        print(f"Order Added: {record['product']}, Order Date: {format_date(record['created_at'], 'DD-MM-YYYY')}")

```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rkj6124/bito-test/pull/7#issuecomment-2557224705>#d1f52a</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged